### PR TITLE
Add mage packaging build and type filters

### PR DESCRIPTION
**Pull request by graphaelli**
eg , build just docker images using existing binaries with:

```
SKIP_BUILD=true SNAPSHOT=true PLATFORMS="linux/amd64" TYPES=docker mage -v package
```

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -115,18 +118,53 @@ func dockerConfigFileParams() mage.ConfigFileParams {
...
+func filterPackages(types string) {
+	var packages []mage.OSPackageArgs
+	keep := keepPackages(strings.Split(types, " "))
+	for _, p := range mage.Packages {
+		for _, t := range p.Types {
+			if _, ok := keep[t]; !ok {
+				continue
+			}
+			packages = append(packages, p)
+			break
```
Comment (jalvz) [inline magefile.go]: why the `break`? can't keep 2 types of the same package or am i reading wrong?

**Comment by graphaelli** [comment]
> maybe add the example of your comment above to the readme, or as a code comment

Something other than https://github.com/elastic/apm-server/pull/3003/files#diff-3db01cb68c42f27f6422c65fa1175e23R151-R152?

File: magefile.go (hunk 1 above)
Comment (graphaelli) [reply magefile.go]: right, there are many types per package but you want just one package once the type matches.  The consequence is that, in our case, if you pick `rpm` you also get `deb`.  I thought about filtering that down further but it didn't seem work it when really all we're after for now is 'build docker images' or 'build linux distro packages'

**Comment by graphaelli** [comment]
transient errors pulling from github, merging

**Comment by graphaelli** [comment]
fyi @andrewkroh

**Pull request merged**